src/main/java/org/commcare/util/CommCarePlatform.java

```diff
@@ -102,9 +111,11 @@ public Entry getEntry(String entryId) {
 
     public void setProfile(Profile p) {
         this.profile = p.getID();
+        this.cachedProfile = p;
     }
 
     public void registerSuite(Suite s) {
+        installedSuites.add(s);
```

Comment:
In `getInstalledSuites()` we are using `installedSuites.addElement` instead of `installedSuites.add`. Is there a catch here due to which one needs to synchronized and other not!

Reply:
No, just an oversight on my part, I don't think that synchronicity would be an issue here but changing this for consistency.